cmdstanpy/cmds.py

```diff
@@ -147,11 +147,13 @@ def sample(
             cores, cpu_count()))
         cores = cpu_count()
     runset = RunSet(args=args, chains=chains)
-    tp = ThreadPool(cores)
-    for i in range(chains):
-        tp.apply_async(do_sample, (runset, i))
-    tp.close()
-    tp.join()
+    try:
```

Comment:
I don't understand why there's a try/finally here w/out exception handling.

Reply:
If that functions fails, it will close the Threadpool and then throw an Exception.

Reply:
got it!  adding this logic as part of PR #40